### ISSUE TITLE
DI extension methods for configuring HTTPS in Builder namespace

### ISSUE THREAD

**Issue by henkmollema** [aspnet/BasicMiddleware#328]
The `AddHttpsRedirection` and `AddHsts` extension methods live in the `Microsoft.AspNetCore.Builder` namespace rather than in usual `Microsoft.Extensions.DependencyInjection` namespace. Is this intentional? 

See:
https://github.com/aspnet/BasicMiddleware/blob/f263e72389fe4468c1c353d2b97ea82704001c31/src/Microsoft.AspNetCore.HttpsPolicy/HstsServicesExtensions.cs#L8-L13

https://github.com/aspnet/BasicMiddleware/blob/f263e72389fe4468c1c353d2b97ea82704001c31/src/Microsoft.AspNetCore.HttpsPolicy/HttpsRedirectionServicesExtensions.cs#L8-L13

**Comment by Tratcher** [comment]
No, it's likely a copy-paste error. However, since you're likely to have both namespaces in Startup.cs the confusion should be limited. Unfortunately it's too late to fix this for 2.1.

**Comment by henkmollema** [comment]
I agree it's no big deal. I stumbled upon it when refactoring some DI configuration to my own extension method and it required me to import the builders namespace. Thanks anyways.

**Comment by jkotalik** [comment]
All other projects use Microsoft.AspNetCore.Builder. See: https://github.com/aspnet/BasicMiddleware/blob/dev/src/Microsoft.AspNetCore.HostFiltering/HostFilteringServicesExtensions.cs#L8 and https://github.com/aspnet/BasicMiddleware/blob/dev/src/Microsoft.AspNetCore.ResponseCompression/ResponseCompressionServicesExtensions.cs

**Comment by Tratcher** [comment]
No, it's inconsistant: https://github.com/aspnet/StaticFiles/blob/cc675047ac81dd8a50e8a77620b4c5ff5a794b8b/src/Microsoft.AspNetCore.StaticFiles/DirectoryBrowserServiceExtensions.cs#L6
https://github.com/aspnet/Mvc/blob/d9f035ad7c4693c314e293f21bfef7ed002dc8e2/src/Microsoft.AspNetCore.Mvc/MvcServiceCollectionExtensions.cs#L13

**Comment by jkotalik** [comment]
Alright, then we should consider switching the namespace on these in the 3.x timeline?
